repository/src/lib.rs

```diff
@@ -267,10 +268,71 @@ impl<T: RawRepository> DistributedRepository<T> {
     /// and update the corresponding `a-#` branch to it
     pub async fn approve(
         &mut self,
-        _agenda_commit_hash: &CommitHash,
-        _proof: Vec<TypedSignature<Agenda>>,
+        agenda_commit_hash: &CommitHash,
+        proof: Vec<TypedSignature<Agenda>>,
     ) -> Result<CommitHash, Error> {
-        unimplemented!()
+        // Check if the agenda branch is rebased on top of the `finalized` branch.
+        let last_header_commit = self.raw.locate_branch(FINALIZED_BRANCH_NAME.into()).await?;
+        let agenda_branch_name = self.raw.get_branches(*agenda_commit_hash).await?;
+        let agenda_branch_name = agenda_branch_name.get(0).unwrap();
```

Comment:
no `unwrap` here. This is just an argument error

Reply:
fixed.